src/Cli/dotnet/Commands/Run/Api/RunApiCommand.cs

```diff
@@ -0,0 +1,100 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Immutable;
+using System.CommandLine;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.DotNet.Cli.Commands.Run.Api;
+
+/// <summary>
+/// Takes JSON from stdin lines, produces JSON on stdout lines, doesn't perform any changes.
+/// Can be used by IDEs to see the project file behind a file-based program.
+/// </summary>
+internal sealed class RunApiCommand(ParseResult parseResult) : CommandBase(parseResult)
+{
+    public override int Execute()
+    {
+        for (string line; (line = Console.ReadLine()) != null;)
```

Comment:
I haven't seen this kind of pattern before, is this a norm for apps which communicate using json over stdio? To essentially put each message on its own line?

Reply:
I don't have particular concerns about the current solution, just trying to make sure I understand the design. Thanks!

Reply:
I'd definitely consider JSON-RPC as it is a standard for this kind of stuff.

Reply:
When I did the out-of-proc build host stuff, I implemented my own RPC via single-line as well. That said, it's a bit unclear to me why we're needing multiple lines like this.

Reply:
> That said, it's a bit unclear to me why we're needing multiple lines like this.

If you expect IDE is going to make only one-off requests (i.e., run new `dotnet run-api` process for each project separately), we can definitely simplify this.

Reply:
If there are no preferences, should we proceed with the current state for starters? If needed, we can change the API later (the `run-api` command is hidden, so I think we are free to change it).

Reply:
I personally have no concerns about the current state. I don't see any reason the command shouldn't handle sequenced requests and responses, even if IDE doesn't make use of that capability.

If we found that spinning a new subprocess to make a project, each time `#:`s are edited or under other various "relatively infrequent" conditions (as compared to updates of ordinary source code), is problematic, then we could start keeping the subprocess alive for some period of time, and making use of this queueing capability.

Reply:
> I'd definitely consider JSON-RPC as it is a standard for this kind of stuff.

@MiYanni Does this comment mean you want me to switch to json-rpc before merging this? There would be some questions like if we want a full compliance with the standard, then we probably need some library (which also means ship a new DLL along the sdk, and likely a new DLL load on the IDE side). Or we can implement this manually like Jason did for out-of-proc (Jason, can you point me to your implementation?). But perhaps we can follow up later since this is an internal API between SDK and IDE.

Reply:
https://github.com/dotnet/roslyn/blob/main/src/Workspaces/MSBuild/BuildHost/Rpc/RpcServer.cs